src/omv/ports/mimxrt/omv_spi.c

```diff
@@ -145,46 +227,64 @@ int omv_spi_transfer_start(omv_spi_t *spi, omv_spi_transfer_t *xfer) {
 }
 
 int omv_spi_transfer_abort(omv_spi_t *spi) {
-    LPSPI_MasterTransferAbortEDMA(spi->inst, &spi->descr_master_edma);
+    if (spi->dma_flags & (OMV_SPI_DMA_NORMAL | OMV_SPI_DMA_CIRCULAR)) {
+        LPSPI_MasterTransferAbortEDMA(spi->inst, &spi->descr_master_edma);
+    }
+    LPSPI_MasterTransferAbort(spi->inst, &spi->descr_master);
+    LPSPI_MasterInit(spi->inst, &spi->config_backup, BOARD_BOOTCLOCKRUN_LPSPI_CLK_ROOT);
```

Comment:
Why do we need to reconfigure the SPI on abort ?

Reply:
Abort leaves the SPI bus broken as it hits the RESET bit in the register... and after pulling my hair out on why it wasn't working I read the docs and it says that after abort you have to reinit the spi bus.

Reply:
Can we try to clear that bit ? If this can be avoided we don't need to duplicate the config, where does it say that in the datasheet ?

Reply:
I tried. I really did. If you abort and don't reset the SPI hardware it breaks. It will not work anymore.

I went as far as implementing all the code in the abort function without the reset... but, no, once you clear the DMA TX/RX registers in the SPI bus you have to reset the whole module.

Reply:
It says this about the abort function in the driver api header file.

Reply:
It was a last resort thing to include the whole config struct again. I did not want to do that. But, I spent a day trying to do a clean approach and got nowhere as the module will not work after abort unless you reinit it.

Reply:
>  It says this about the abort function in the driver api header file.

Which file ? Link ?

Do we need to call both `LPSPI_MasterTransferAbortEDMA` and `LPSPI_MasterTransferAbort` for a DMA transfer ? Could that be the issue ?

Reply:
This can't be right because in the current driver it works without re-initializing the bus and there's an abort, I've tested this with vospi + FLIR.

Reply:
Abort calls LPSPI_RESET.

```
/*!
 * brief LPSPI master abort transfer which uses an interrupt method.
 *
 * This function aborts a transfer which uses an interrupt method.
 *
 * param base LPSPI peripheral address.
 * param handle pointer to lpspi_master_handle_t structure which stores the transfer state.
 */
void LPSPI_MasterTransferAbort(LPSPI_Type *base, lpspi_master_handle_t *handle)
{
    assert(handle != NULL);

    /* Disable interrupt requests*/
    LPSPI_DisableInterrupts(base, (uint32_t)kLPSPI_AllInterruptEnable);

    LPSPI_Reset(base);

    handle->state                = (uint8_t)kLPSPI_Idle;
    handle->txRemainingByteCount = 0;
    handle->rxRemainingByteCount = 0;
}
```

Which says:

```
/*!
 * brief Restores the LPSPI peripheral to reset state. Note that this function
 * sets all registers to reset state. As a result, the LPSPI module can't work after calling
 * this API.
 * param base LPSPI peripheral address.
 */
void LPSPI_Reset(LPSPI_Type *base)
{
    /* Reset all internal logic and registers, except the Control Register. Remains set until cleared by software.*/
    base->CR |= LPSPI_CR_RST_MASK;

    /* Software reset doesn't reset the CR, so manual reset the FIFOs */
    base->CR |= LPSPI_CR_RRF_MASK | LPSPI_CR_RTF_MASK;

    /* Master logic is not reset and module is disabled.*/
    base->CR = 0x00U;
}
```

You only abort the DMA in the current code:

```
int omv_spi_transfer_abort(omv_spi_t *spi) {
    LPSPI_MasterTransferAbortEDMA(spi->inst, &spi->descr_master_edma);
    return 0;
}
```

This is not applicable when doing non-DMA transfers.

Reply:
> This is not applicable when doing non-DMA transfers.

Do we need to recover the bus if a blocking/IT transfer fails ? If so then still it should be :

```c
    } else {
       LPSPI_MasterTransferAbort(spi->inst, &spi->descr_master);
       LPSPI_MasterInit(spi->inst, &spi->config_backup, BOARD_BOOTCLOCKRUN_LPSPI_CLK_ROOT);
}
```

Reply:
This might be better.